optuna/integration/allennlp.py

```diff
@@ -9,17 +10,33 @@
 import optuna
 from optuna._experimental import experimental
 from optuna._imports import try_import
+from optuna import load_study
+from optuna import Trial
+
 
 with try_import() as _imports:
     import allennlp
     import allennlp.commands
     import allennlp.common.util
-    from allennlp.training import EpochCallback
 
 if _imports.is_successful():
     import _jsonnet
+    from allennlp.training import EpochCallback
 else:
-    EpochCallback = object  # NOQA
+
+    class EpochCallback:  # type: ignore
```

Comment:
We should explain why `# type: ignore` as I don't think it's evident at a glance.

Reply:
You're right. I added some notes in 5ef26c8.